include/zephyr/net/wifi_mgmt.h

```diff
@@ -389,10 +389,10 @@ enum wifi_conn_status {
  * in the disconnect result event for detailed reason.
  */
 enum wifi_disconn_reason {
-	/** Unspecified reason */
-	WIFI_REASON_DISCONN_UNSPECIFIED = WIFI_STATUS_DISCONN_FIRST_STATUS,
 	/** Disconnected due to user request */
 	WIFI_REASON_DISCONN_USER_REQUEST,
+	/** Unspecified reason */
+	WIFI_REASON_DISCONN_UNSPECIFIED = WIFI_STATUS_DISCONN_FIRST_STATUS,
```

Comment:
instead of this mishmash, should we instead add a check in the shell to deal with this?

Reply:
Actually this change will help to use WIFI_REASON_DISCONN_USER_REQUEST of Enum wifi_disconn_reason. Currently it is unused. By changing to 0 this will place disconnection request successfully.

Reply:
This looks very weird now, the `WIFI_REASON_DISCONN_USER_REQUEST` gets value 0, and the unspecified reason gets a value after the last connection status value. This looks very confusing, why would we need to do anything like this in the first place? Why not check this thing in shell as suggested by @krish2718

Reply:
When user tries to disconnect using wifi disconnect command the disconnection result is getting WIFI_REASON_DISCONN_UNSPECIFIED only. The idea is to use WIFI_REASON_DISCONN_USER_REQUEST for user to disconnect. For every non 0 value, disconnection result will be show as failed. For that making WIFI_REASON_DISCONN_USER_REQUEST as a 0.

Still if don't want to change that, I can make change in wifi shell condition, and check if it's WIFI_STATUS_CONN_LAST_STATUS than disconnect request will be successful.

Reply:
> Still if don't want to change that, I can make change in wifi shell condition, and check if it's WIFI_STATUS_CONN_LAST_STATUS than disconnect request will be successful.

Yes, I think this is a better solution as it is causing less confusion to anyone reading the code.